packages/babel-helpers/src/helpers.js

```diff
@@ -413,6 +416,29 @@ helpers.interopRequireWildcard = template(`
   })
 `);
 
+helpers.specRequireInterop = template(`
+  (function (obj) {
+    if (obj && obj.__esModule) {
+      return obj;
+    } else {
+      var newObj = Object.create(null, {
+          default: {
+            value: obj,
+            writable: true,
+            enumerable: true
+          },
+          __esModule: {
```

Comment:
If we assume that nothing actually checks for `__esModule` equality, it could be set to `value: 'spec'`; but it's probably not a good idea to add even more nonstandard keys...